scRNA-seq-advanced/setup/intro-diff-expression-setup.Rmd

```diff
@@ -0,0 +1,633 @@
+
+A good question we could ask is if there are differences in gene expression patterns in the Myoblast population between ARMS and ERMS patients. 
+If we want to ask that question, then we need to have replicates, preferably at least 3 samples for each type of patient, which we do! 
+
+Let's go ahead and try it! 
+
+## Differential Expression between conditions 
+
+### Pseudo-bulking 
+
+Before we can do DE analysis to compare the gene expression profiles of myoblasts in ARMS vs. ERMS patients, we will need to "pseudo-bulk" the gene counts. 
+Pseudo-bulking creates a new counts matrix that contains the total counts across all cells of a given label (e.g. cell type) for each sample. 
+This allows us to use single-cell resolution to define the labels and sum gene counts across groups of cells containing the same label, and avoids counting each cell as its own replicate. 
+
+Below are a few reasons why pseudo-bulking is helpful: 
+
+- Produces larger counts which are more amenable to standard normalization and differential expression methods used by bulk RNA-sequencing. 
+-  Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
+- Masks variance within a sample to highlight variance across samples.
+
+Let's start with the simple comparison of looking at one cell type and comparing across ERMS and ARMS subtypes. 
+When we do this, we will first want to pseudo-bulk our dataset to group our cells by cell type and by sample.
+Then we can subset the pseudo-bulked dataset to just contain our cell type of interest before proceeding. 
+
+```{r}
+# first let's subset our coldata to only have the columns we care about in pseudobulking 
+pseudobulked_coldata <- colData(rms_sce)[, c("celltype_broad", "batch")]
+
+# this creates a new SCE object that contains the pseudobulked counts across the provided groups 
+pseudobulked_sce <- scater::aggregateAcrossCells(rms_sce, 
+                                                  id = pseudobulked_coldata)
+
+pseudobulked_sce
+```
+
+```{r}
+# only 37 columns in the counts assay 
+counts(pseudobulked_sce)[1:10, ]
+```
+
+Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+
+```{r}
+# note the new addition of the column with number of cells per group 
+colData(pseudobulked_sce)
+
+```
+
+Now we filter to just the sub type of cells that we are interested in. 
+
+```{r}
+tumor_myoblast_sce <- pseudobulked_sce[, pseudobulked_sce$celltype_broad == "Tumor_Myoblast"]
+table(tumor_myoblast_sce$celltype_broad)
+```
+
+We should now see that there are only 6 samples left that all have the cell type `Tumor_Myoblast`. 
+
+Before we can perform DGE analysis, we want to do some quality control like filtering out any groups in the pseudo-bulked dataset that may have a low number of cells. 
+
+```{r}
+# filter by number of cells 
+tumor_myoblast_sce <- tumor_myoblast_sce[, tumor_myoblast_sce$ncells >= 10]
```

Comment:
No there aren't any, but yes we want to show it. Also we could move it to before we filter to just the myoblast population since the filtering we should do across the whole pseudobulked sce.